src/Microsoft.DotNet.Interactive.CSharp/CSharpKernel.cs

```diff
@@ -221,27 +221,35 @@ public async Task HandleAsync(SubmitCode submitCode, KernelInvocationContext con
 
             if (!context.CancellationToken.IsCancellationRequested)
             {
+                var diagnostics = ImmutableArray<CodeAnalysis.Diagnostic>.Empty;
```

Comment:
This grabs the diagnostics (whether they be errors or warnings) and reports them.

The formatting of error text is the same as before, however that text always goes to `DiagnosticsProduced` instead of being attached to the exceptions in `context.Fail(...)`